packages/back-end/src/scim/users/getUser.ts

```diff
@@ -0,0 +1,71 @@
+import { Response } from "express";
+import { ScimGetRequest } from "../../../types/scim";
+import { getUserById } from "../../services/users";
+
+export async function getUser(req: ScimGetRequest, res: Response) {
+  const userId = req.params.id;
+
+  const user = await getUserById(userId);
+
+  if (!user) {
+    return res.status(404).json({
+      schemas: ["urn:ietf:params:scim:api:messages:2.0:Error"],
+      detail: "User ID does not exist",
+    });
+  }
+
+  const org = req.organization;
+
+  const orgUser = org.members.find((member) => member.id === user?.id);
+
+  // TODO: Create a function to map Growthbook users to SCIM users that we can use here and in listUsers
+  if (!orgUser) {
```

Comment:
I'm wondering if we need to change the response based on whether or not the organization is self-hosted or cloud.

For cloud users, this could potentially leak user information that this org shoudn't have access to - if someone was able to get a user's ID (or guess it) they could hit this and get the user's email and name. Probably obscure, but feasible.

I think this would be fine for self-hosted orgs, but if the org is on a cloud plan, we may want to just return a 404. Though, that might cause Okta's automated test to fail, in that case, we could return a 200 with a ListResponse schema. 

```ts
 return res.status(200).json({
    schemas: ["urn:ietf:params:scim:api:messages:2.0:ListResponse"],
    totalResults: 0,
    Resources: resources,
    startIndex: 1,
    itemsPerPage: 20,
  });```

Reply:
I don't think a `ListResponse` would work here because SCIM requires the response for `getUser` to be a single `ScimUser` I'm going to try to incorporate the idea we discussed where we explicitly store deletedUsers on the org so that we don't need different approaches for cloud and self-hosted